subprojects/packagefiles/xxhash/meson.build

```diff
@@ -55,7 +55,10 @@ if get_option('prefetch-dist') > 0
     c_args += '-DXXH_PREFETCH_DIST=@0@'.format(get_option('prefetch-dist'))
 endif
 
-if not get_option('inline-hints').auto()
+if cc.get_id() == 'gcc' and cc.version().version_compare('>=12.0.0') and get_option('optimization') == 'g'
+    c_args += '-DXXH_NO_INLINE_HINTS=1'
+    message('Ignoring value of inline-hints, see https://github.com/Cyan4973/xxHash/pull/720')
+elif not get_option('inline-hints').auto()
```

Comment:
Does this work if I pass in options with CFLAGS or C_ARGS?

Reply:
Hmmm, no inline hints, how will that affect the build quality?

Reply:
Greg, look here: https://github.com/Cyan4973/xxHash/blob/dev/xxhash.h#L1677. That is the behavior if left unspecified.

Reply:
CFLAGS should get passed down to the build of xxhash. @eli-schwartz can correct me if I am wrong.

Reply:
We don't set this particular option in HSE, but we do set `inline-all`

Reply:
If you pass -Og in via CFLAGS then Meson doesn't sync that with `get_option('optimization')` and this meson.build will simply progress on to check the inline-hints option.

So in the case you used CFLAGS here, you'd want to make sure you set inline-hints yourself, or leave it at the default autodetection which may do anything, but won't take compiler versions into account.